altsrc/json_source_context.go

```diff
@@ -195,7 +196,7 @@ func jsonGetValue(key string, m map[string]interface{}) (interface{}, error) {
 	keys := strings.Split(key, ".")
 	for ix, k := range keys {
 		if ret, ok = working[k]; !ok {
-			return ret, fmt.Errorf("missing key %q", key)
+			return ret, syscall.ENOENT
```

Comment:
Rather than returning a value that already has meaning attached to it, I think this should instead be a stable error that's declared at the package level, e.g. `errMissingKey`, which is then checked via `errors.Is` rather than an equality check.

Reply:
@meatballhat I've made the change